internal/statemachine/rootfs.go

```diff
@@ -0,0 +1,20 @@
+package statemachine
+
+import "github.com/canonical/ubuntu-image/internal/commands"
+
+// RootfsStateMachine embeds ClassicStateMachine and enable re-implementing
+// SetCommonOpts to stop the Classic workflow once the rootfs is ready
+type RootfsStateMachine struct {
+	ClassicStateMachine
+}
+
+// SetCommonOpts is forcing the RootfsStateMachine to stop at a specific state
+// so only the rootfs is generated
+func (s *RootfsStateMachine) SetCommonOpts(commonOpts *commands.CommonOpts,
+	stateMachineOpts *commands.StateMachineOpts) {
+	s.StateMachine.SetCommonOpts(commonOpts, stateMachineOpts)
+
+	// Make sure to stop the ClassicStateMachine at a specific state
+	s.StateMachine.stateMachineFlags.Thru = preseedClassicImageState.name
```

Comment:
I agree and as mentioned in another comment I have some ideas to implement it but I also agree this should be in another PR.

My current idea is to add a "meta" field to the stateFunc struct to associate the stateFunc to a meta state. When the runner (the for loop) is executing states, it could stop when a meta state is done this way.

Thinking about the state machine, I also noticed that the concept of state machine in our situation may be a bit "overkill" because:
- we calculate states at the beginning and then execute them one after another, without ever going back to a previous state
- our state machine is a single branch. We never "branch" in another direction based on the result of a state.

I do not think this is bothering us and I feel like it could be useful at some point (maybe when dealing with multi-layer FS for the installer?) but I do not know if we should keep designing features around this concept.

Back to the "meta state" subject: if we consider our state machine as a simple list, then we only need to mark the last state of a meta state as "\<metastate\>-end". As you mentioned, during a refactoring this could change, but we would have to associate added/removed states to the meta-state anyway. But at least this way the stateFunc is holding the association and it is obvious it should be defined when adding a stateFunc (we could even enforce it by moving the stateFunc in a dedicated package and only enable creating stateFuncs through a `New()` method, forcing non-empty values for name, func and metaState).